[drogon] Update to 1.4.1

an-tao opened the pull request.

an-tao committed:
[drogon] Update to 1.4.1

an-tao committed:
Update patch

JonLiu1993 commented on ports/drogon/CONTROL:
```diff
@@ -1,8 +1,5 @@
 Description:Drogon: A C++14/17 based HTTP web application framework running on Linux/macOS/Unix/Windows
-Build-Depends: trantor, zlib, jsoncpp, libmariadb (!osx), libmariadb[iconv] (osx), libpq, sqlite3, brotli, libuuid (!windows)
-
-Feature: ctl
```
Hi @an-tao ,Why did you delete this feature?

an-tao replied:
I think drogon_ctl is needed in almost all scenarios. 
Actually, some people complained to me that drogon_ctl is missing after installing drogon in the latest version of vcpkg.

JonLiu1993 replied:
Ok that's great

ras0219-msft replied:
I think we should keep this as a feature, but make it a default:
```
Default-Features: ctl
```
This is useful for cases like cross-compiling to android or ios where creating an executable may not be possible.

JonLiu1993 commented on ports/drogon/portfile.cmake:
```diff
@@ -1,46 +1,42 @@
-        vcpkg.patch
-        resolv.patch
-)
+  OUT_SOURCE_PATH
+  SOURCE_PATH
+  REPO
+  an-tao/drogon
+  REF
+  v1.4.1
+  SHA512
+  8611c18e65229095f5443f10c87e91593d619bc3a2d47da6d19d501a64be1aba12a813e44ccfc1c10179e5b0fa10121e8806955c0091258992501b88fa50d939
+  HEAD_REF
+  master
+  PATCHES
+  vcpkg.patch
+  resolv.patch)
```
@an-tao ，Please restore the format here.

JonLiu1993 commented on ports/drogon/portfile.cmake:
```diff
@@ -1,46 +1,42 @@
-        -DBUILD_EXAMPLES=OFF
-        ${FEATURE_OPTIONS}
-)
+vcpkg_configure_cmake(SOURCE_PATH
+                      ${SOURCE_PATH}
+                      PREFER_NINJA
+                      OPTIONS
+                      -DBUILD_EXAMPLES=OFF)
```
@an-tao ，Please keep the format here the same as before.

an-tao committed:
Format

an-tao commented:
@JonLiu1993 If some dependencies update after the PR and break the build of drogon, what can I do to avoid this situation?

JonLiu1993 commented:
> @JonLiu1993 If some dependencies update after the PR and break the build of drogon, what can I do to avoid this situation?

I'll fix it

JonLiu1993 committed:
update version

an-tao commented:
> If a new problem arises in response to this situation, we will solve it, you know this cannot be avoided

I see, thanks for your reply.

ras0219-msft reviewed (changes requested):
Thanks for the PR! LGTM once the feature is moved to default.

an-tao committed:
Revert the ctl feature

JonLiu1993 committed:
update version

ras0219-msft committed:
[libpq] Fix static linkage

ras0219-msft committed:
[drogon] Revert direct postgresql library linkage

ras0219-msft commented:
Hi @an-tao, thanks for the PR!

Obviously needing to directly link lower level postgresql libraries is not great, so I've fixed the underlying libpq port and reverted that part of the drogon patches here: https://github.com/ras0219-msft/vcpkg/commits/drogon. Could you please pull those additional commits into this PR?

Additionally, I noticed that drogon performs some automatic detection with a boost polyfill fallback -- this hidden optional dependency causes a bunch of issues in a package manager so I've taken the simplest route for now and ensured the build will fail if that path would be taken.

an-tao commented:
> Hi @an-tao, thanks for the PR!
> 
> Obviously needing to directly link lower level postgresql libraries is not great, so I've fixed the underlying libpq port and reverted that part of the drogon patches here: https://github.com/ras0219-msft/vcpkg/commits/drogon. Could you please pull those additional commits into this PR?
> 
> Additionally, I noticed that drogon performs some automatic detection with a boost polyfill fallback -- this hidden optional dependency causes a bunch of issues in a package manager so I've taken the simplest route for now and ensured the build will fail if that path would be taken.

Thank you very much, I have added them to this PR, you solved a very annoying problem for me.

ras0219-msft commented:
Thanks so much for the PR!
